### PR TITLE
Slawomir/auth

### PR DESCRIPTION
**Pull request by thewilkupl**
This pull request contains:

- implemented login/register form with validation
- implemented globalStyles.js (Styled components)

### CODE REVIEW

File: frontend/src/components/Auth/FormFormik.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import React from "react";
+import React, { useState } from "react";
 import { Formik, Field, Form, ErrorMessage } from "formik";
 import * as Yup from "yup";
 import styled from "styled-components";
 
 import { MdEmail } from "react-icons/md";
```
Comment (Willaiem) [inline frontend/src/components/Auth/FormFormik.tsx:6]: Add aliases for these icons for better readability i.e. MdEmail -> EmailIcon

File: frontend/src/components/Auth/FormFormik.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-import { RiLockPasswordFill } from "react-icons/ri";
+import { AiFillLock, AiFillUnlock } from "react-icons/ai";
```
Comment (Willaiem) [inline frontend/src/components/Auth/FormFormik.tsx:8]: Add aliases for these icons for better readability i.e. MdEmail -> EmailIcon

File: frontend/src/components/Auth/FormFormik.tsx [hunk 3]
```diff
@@ -13,6 +13,8 @@ type FormProps = {
 };
 
 const FormFormik: React.FC<FormProps> = ({ isSignUp }) => {
+  const [showPassword, setShowPassword] = useState(false);
+
```
Comment (Willaiem) [inline frontend/src/components/Auth/FormFormik.tsx:17]: Change "showPassword" to yes/ no type "question" for booleans (e.g. isPasswordVisible)

File: frontend/src/components/Auth/FormFormik.tsx [hunk 4]
```diff
@@ -136,11 +144,17 @@ const Wrapper = styled.div`
...
+  }
 `;
 
 const Search = styled.div`
   padding: 0.5rem;
-  margin: 0.5rem;
+  margin: 0.2rem;
   position: relative;
   display: flex;
   align-items: center;
```
Comment (Willaiem) [inline frontend/src/components/Auth/FormFormik.tsx:178]: Move the styles to separated files i.e: Search to Search.styles.ts.

File: frontend/src/components/Auth/FormFormik.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+import * as Yup from "yup";
+import styled from "styled-components";
+
+import { MdEmail } from "react-icons/md";
+import { RiLockPasswordFill } from "react-icons/ri";
+
+type FormProps = {
+  isSignUp: boolean;
+};
+
```
Comment (Willaiem) [inline frontend/src/components/Auth/FormFormik.tsx:14]: Move all types declarations to separated files i.e FormProps -> FormFormik.types.ts

**Review by Willaiem — CHANGES_REQUESTED** [review]
I added the suggestions before merging.

File: frontend/src/components/Auth/FormFormik.tsx (hunk 5 above)
Comment (thewilkupl) [reply frontend/src/components/Auth/FormFormik.tsx:18]: I think co-locate styles in the component are fine for now. Seems I don’t see a lot of benefit of separating them.

File: frontend/src/components/Auth/Auth.tsx [hunk 6]
```diff
@@ -5,8 +5,8 @@ import { Link } from "react-router-dom";
 import styled from "styled-components";
 
 //ICONS
-import { FcGoogle } from "react-icons/fc";
```
Comment (Willaiem) [inline frontend/src/components/Auth/Auth.tsx]: I think comments are unnecessary when variables are self-explanatory.

File: frontend/src/components/Auth/FormFormik.tsx (hunk 5 above)
Comment (Willaiem) [reply frontend/src/components/Auth/FormFormik.tsx:18]: That's right for this moment, but when styles will grow, then separating them will be inevitable to not clutter up the components. 
Personally, I would keep things organized.

**Review by Willaiem — APPROVED** [review]
Everything looks good 👌

**Pull request merged**